app/controllers/projects_controller.rb

```diff
@@ -343,6 +343,6 @@ def return_context
     render_404 && return if @project.blank?
     return unless @project.deleted?
     flash[:alert] = 'The project you requested had been deleted.'
-    redirect_to user_path(@user)
+    check_redirect "/#{@user.username}"
```

Comment:
I think it wouldn't give warning if we use rails to define our routes rather than our own methods like `urlbase`. You are working on routes, I hope you will figure it out? I guess this can wait until then.